File: src/main/kotlin/com/geonyeong/insurance/entity/GuideFileEntity.java
package com.geonyeong.insurance.entity;

import com.geonyeong.insurance.dto.GuideIdDto;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.ColumnDefault;

import javax.persistence.*;

@Entity
@Table(name = "guide_file")
@IdClass(GuideIdDto.class)
@Getter
@Setter
public class GuideFileEntity {
    @Id
    private @Column(name = "guide_id") Long guideId;
    @Id
    private @Column(name = "guide_no") Long guideNo;
    @Id
    private @Column(name = "no") Long no;
    private @Column(name = "route") String route;
    private @Column(name = "title") String title;
    @ColumnDefault("NULL")
    private @Column(name = "preview") String preview;
}

File: src/main/kotlin/com/geonyeong/insurance/entity/GuideFileEntity.kt
package com.geonyeong.insurance.entity

import com.fasterxml.jackson.annotation.JsonProperty
import com.geonyeong.insurance.dto.GuideIdDto
import org.hibernate.annotations.ColumnDefault
import java.io.Serializable
import javax.persistence.*

@Entity
@Table(name = "guide_file")
@IdClass(GuideIdDto::class)
data class GuideFileEntity(
        @Id
        @Column(name = "guide_id")
        @field:JsonProperty("guideId")
        var guideId: Long?,
        @Id
        @Column(name = "guide_no")
        @field:JsonProperty("guideNo")
        var guideNo: Long?,
        @Id
        @Column(name = "no")
        @field:JsonProperty("no")
        var no: Long?,
        @Column(name = "route")
        @field:JsonProperty("route")
        var route: String?,
        @Column(name = "title")
        @field:JsonProperty("title")
        var title: String?,
        @ColumnDefault("NULL")
        @Column(name = "preview")
        @field:JsonProperty("preview")
        var preview: String?
):Serializable{
        constructor():this(0,0,0,"","","")
}